### ISSUE TITLE
WARNING: CouchbaseLite: Failed to rebuild views

### ISSUE THREAD

**Issue by kevinkirkup** [couchbase/couchbase-lite-ios#690]
Application is triggering a WARNING on start which ends up causing the Map block to never run.
The database does contain data. The ```createAllDocuments()``` method works as expected.

14:18:09.744| WARNING: CouchbaseLite: Failed to rebuild views (products/byProducts): 593
14:18:09.759| WARNING: Failed to update view index: 593

```
14:18:09.743| WARNING: Exception caught in map block of view 'products/byProducts':
	*** -[__NSPlaceholderArray initWithObjects:count:]: attempt to insert nil object from objects[0]
	2   CoreFoundation                      0x00000001017d3478 -[__NSPlaceholderArray initWithObjects:count:] + 360
	3   CoreFoundation                      0x0000000101832384 +[NSArray arrayWithObjects:count:] + 52
	4   REDACTED                            0x00000001004be680 __42+[CouchDatabase productsViewWithDatabase:]_block_invoke + 672
	5   REDACTED                            0x00000001004eb2f8 __51-[CBLDatabase(ViewIndexing) updateIndexes:forView:]_block_invoke + 4989
	6   REDACTED                            0x00000001004e1579 -[CBLDatabase(Internal) _inTransaction:] + 113
	7   REDACTED                            0x00000001004e9e36 -[CBLDatabase(ViewIndexing) updateIndexes:forView:] + 170
	8   REDACTED                            0x00000001004e9cd5 -[CBLView(Internal) updateIndex] + 85
	9   REDACTED                            0x000000010052090d -[CBLDatabase(Views) queryViewNamed:options:lastSequence:status:] + 170
	10  REDACTED                            0x000000010051dada -[CBLQuery run:] + 127
	11  REDACTED                            0x00000001004c1703 _TFC13REDACTED8Database9queryViewfS0_FTCSo7CBLView7handlerFCSo11CBLDocumentT__T_ + 355
	12  REDACTED                            0x00000001004c1363 _TFC13REDACTED8Database8productsfS0_FT_GSaC11REDACTEDKit7Product_ + 339
	13  REDACTED                            0x00000001004c0056 _TFC13REDACTED8Database19replicationProgressfS0_FCSo20NSNotificationCenterT_ + 790
	14  REDACTED                            0x00000001004c00d9 _TToFC13REDACTED8Database19replicationProgressfS0_FCSo20NSNotificationCenterT_ + 57
	15  CoreFoundation                      0x00000001018de54c __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
	16  CoreFoundation                      0x00000001017dca04 _CFXNotificationPost + 2484
	...
14:18:09.744| WARNING: CouchbaseLite: Failed to rebuild views (products/byProducts): 593
14:18:09.759| WARNING: Failed to update view index: 593
```

Backtrace:
```
* thread #1: tid = 0xf599ed, 0x00000001015a7b8a libobjc.A.dylib`objc_exception_throw, queue = 'com.apple.main-thread', stop reason = breakpoint 4.1
    frame #0: 0x00000001015a7b8a libobjc.A.dylib`objc_exception_throw
    frame #1: 0x00000001017d3478 CoreFoundation`-[__NSPlaceholderArray initWithObjects:count:] + 360
    frame #2: 0x0000000101832384 CoreFoundation`+[NSArray arrayWithObjects:count:] + 52
  * frame #3: 0x00000001004be680 REDACTED`__42+[CouchDatabase productsViewWithDatabase:]_block_invoke(.block_descriptor=0x00000001005c5160, doc=0x00007f87b348ead0, emit=0x00007f87b3489c00) + 672 at CouchDatabase.m:40
    frame #4: 0x00000001004eb2f8 REDACTED`__51-[CBLDatabase(ViewIndexing) updateIndexes:forView:]_block_invoke + 4989
    frame #5: 0x00000001004e1579 REDACTED`-[CBLDatabase(Internal) _inTransaction:] + 113
    frame #6: 0x00000001004e9e36 REDACTED`-[CBLDatabase(ViewIndexing) updateIndexes:forView:] + 170
    frame #7: 0x00000001004e9cd5 REDACTED`-[CBLView(Internal) updateIndex] + 85
    frame #8: 0x000000010052090d REDACTED`-[CBLDatabase(Views) queryViewNamed:options:lastSequence:status:] + 170
    frame #9: 0x000000010051dada REDACTED`-[CBLQuery run:] + 127
    frame #10: 0x00000001004c1703 REDACTED`REDACTED.Database.queryView (view=CBLView at 0x00007fff5f73ff60, handler=<unavailable>, self=0x00007f87b3657390)(ObjectiveC.CBLView, handler : (ObjectiveC.CBLDocument) -> ()) -> () + 355 at Database.swift:216
    frame #11: 0x00000001004c1363 REDACTED`REDACTED.Database.products (self=0x00007f87b3657390)() -> Swift.Array<REDACTEDKit.Product> + 339 at Database.swift:200
```

**Comment by snej** [comment]
tl;dr: This is a bug in your app, not in Couchbase Lite.

As the warning says, the exception is thrown by your application's map block. The backtrace shows it's a block in your method `-[CouchDatabase productsViewWithDatabase:]`, source line CouchDatabase.m:40.
